File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int result=0,left=0;
        unordered_map<char,int>m;
        for(int right=0;right<s.size();right++){
            m[s[right]]++;
            while(m[s[right]]>1){
                m[s[left]]--;
                left++;
            }
            result=max(result,right-left+1);
        }
        return result;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left,ans=0,0
        m={}
        for right in range(0,len(s)):
            m[s[right]]=m.get(s[right],0)+1
            while m[s[right]]>1:
                m[s[left]]-=1
                left+=1
            ans=max(ans,right-left+1)
        return ans
        
        